src/adapters/data_models/shortcuts_model/shortcuts_proxy_model.cpp

```diff
@@ -30,19 +30,25 @@ bool ShortcutsProxyModel::lessThan(const QModelIndex& left,
     QString rightName = sourceModel()->data(right, ShortcutRole).toString();
 
     // Sort alphabetically if no filter string is set
-    if(m_filterString.isEmpty())
+    if(!m_filterScorer)
         return leftName < rightName;
 
     // Sort by the highest similarity if filter string is set
-    auto lSimilarity = string_utils::fuzzCompare(leftName, m_filterString);
-    auto rSimilarity = string_utils::fuzzCompare(rightName, m_filterString);
+    auto lSimilarity = string_utils::substringCompare(leftName, m_filterString);
+    if(lSimilarity == 0.0)
+        lSimilarity = m_filterScorer.similarity(leftName.toUcs4());
+
+    auto rSimilarity = string_utils::substringCompare(rightName, m_filterString);
+    if(rSimilarity == 0.0)
+        rSimilarity = m_filterScorer.similarity(rightName.toUcs4());
```

Comment:
Especially for short strings this is quite a bit more efficient, since it doesn't has to repeatedly create bitvectors for the first string.

Reply:
So do I understand it correctly that `rapidfuzz::fuzz::CachedRatio` is a type that contains the `m_filterString` converted to a bitvector, so we are simply saving one conversion on every comparison (thus 50% of the conversions)?

Reply:
And `m_filterScorer.similarity(other)` where the `m_filterScorer` contains the `m_filterString` does the same as `rapidfuzz::fuzz::ratio(m_filterString, other);` right?

Reply:
On a high level the algorithm works in the following way:
```py
pm = BitVector(s1)
for ch2 in s2:
     pm_j = pm[ch2]
     ...
```
So only one of the strings is converted to a bitvector. These bitvectors encode the positions of characters in a string and allow comparing 64 characters in parallel. For long strings (>64 characters) multiple bitvectors need to be chained:
```py
pm = BitVector(s1)
for ch2 in s2:
     for word in ceil(len(s1) / 64)):
         pm_j = pm[ch2][word]
          ...
```
For long texts the conversion to a bitvector does not matter much, since the nested loop takes most of the runtime. However for shorter texts this conversion can take up to 50% of the total runtime.

Reply:
> And m_filterScorer.similarity(other) where the m_filterScorer contains the m_filterString does the same as rapidfuzz::fuzz::ratio(m_filterString, other); right?

Yes exactly.

Reply:
There is another method not as relevant for your usage `MultRatio` which allows encoding multiple short texts in bitvectors and comparing them using simd instructions. This allows e.g. comparing 32 * 8 character texts in parallel using AVX2. It's still in the experimental namespace, since I am not completely sure about the API yet.

Reply:
This reminds me that I REALLY need to write some better documentation for the C++ version at some point :sweat_smile: